### PR TITLE
Add tests for replication

### PR DESCRIPTION
**Pull request by sujeebant**
Adding tests for replication between mock and masq-lib

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -307,7 +307,8 @@ class Masq {
           if (stayConnected) this._storeSessionInfo(json.userAppDbId)
           this.userId = json.userAppDbId
 
-          const db = utils.createPromisifiedHyperDB(this.userId, json.key)
+          const buffKey = Buffer.from(json.key, 'hex')
+          const db = utils.createPromisifiedHyperDB(this.userId, buffKey)
```
Comment (G-Ray) [inline src/index.js]: ok

File: test/masq.test.js [hunk 2]
```diff
@@ -18,14 +19,25 @@ jest.mock('random-access-idb', () =>
...
   const original = require.requireActual('../src/utils')
+  let dbList = {}
   return {
     ...original,
-    dbExists: jest.fn(() => false)
+    dbExists: (name) => {
+      return Promise.resolve(!!dbList[name])
+    },
+    createPromisifiedHyperDB: (name, hexKey) => {
+      dbList[name] = 'db'
```
Comment (G-Ray) [inline test/masq.test.js]: why not store the dbInstance returned by createPromisifiedHyperDB here ?

File: test/masq.test.js [hunk 3]
```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
...
+  }
+  const onChangeMockMasqApp = () => {
+    expect(true).toBe(true)
+    resolvePr2()
+  }
+
+  await logInWithMasqAppMock()
+  const key = '/hello'
+  const value = { data: 'world' }
+  mockMasqApp.watch(masq.userId, '/hello', onChangeMockMasqApp)
```
Comment (G-Ray) [inline test/masq.test.js]: I don't think this is necessary? You just need to focus on the behavior of masq-lib

File: test/masq.test.js [hunk 4]
```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
...
+  done()
+})
+
+test('put/get should put an item and get in Mock Masq App', async () => {
+  expect.assertions(1)
+  await logInWithMasqAppMock()
+
+  let resolvePr
+  const pr = new Promise((resolve) => { resolvePr = resolve })
+  const onChangeMockMasqApp = () => {
```
Comment (G-Ray) [inline test/masq.test.js]: I see no point to do this. Just call `done()` after `mockMasqApp.watch(masq.userId, '/hello', () => done())`, and check the values returned by watch in the callback (before calling done)

File: test/masq.test.js [hunk 5]
```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
...
+  const value = { data: 'world' }
+  mockMasqApp.watch(masq.userId, '/hello', onChangeMockMasqApp)
+  masq.watch('/hello', onChangeMasqLib)
+  await mockMasqApp.put(masq.userId, key, value)
+
+  await Promise.all([pr1, pr2])
+  done()
+})
+
+test('put/get should put an item and get in Mock Masq App', async () => {
```
Comment (G-Ray) [inline test/masq.test.js]: I'm not sure what you want to test here, why do we have a watch called ?

Comment (G-Ray) [reply test/masq.test.js]: I guess you want to wait the data to be syncronized. In this case, can you create a new test file, inside a `describe` called `replication` or something like this

File: test/masq.test.js (hunk 2 above)
Comment (sujeebant) [reply test/masq.test.js]: dbList is only meant to make dbExists functional in tests
We do not need to store the instance here

File: test/masq.test.js (hunk 3 above)
Comment (sujeebant) [reply test/masq.test.js]: It is the only way to make sure that the two-way replication is done correctly in tests.

File: test/masq.test.js (hunk 4 above)
Comment (sujeebant) [reply test/masq.test.js]: I would rather not use async and done at the same time as we would not get an error if done is called before an awaited promise is resolved.

File: test/masq.test.js (hunk 5 above)
Comment (sujeebant) [reply test/masq.test.js]: I want to test the replication.
We have to wait before getting the value to make sure the 2 hyperdb instances are synced after the put.

We could put the last two tests in a separate file but I don't see the point as we have not isolated the other tests. Maybe we should separate tests into login, get/put/delete, watch and replication ?
But I think we can leave them in the same file for now, what do you think ?

File: test/masq.test.js (hunk 4 above)
Comment (G-Ray) [reply test/masq.test.js]: It this is really needed, you can just write:
`onChangeMockMasqApp = () => Promise.resolve()

File: test/masq.test.js (hunk 5 above)
Comment (sujeebant) [reply test/masq.test.js]: Okay !

File: test/masq.test.js [hunk 6]
```diff
@@ -555,3 +578,50 @@ test('should set a watcher', async (done) => {
   masq.watch('/hello', onChange)
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply test/masq.test.js]: I don't think there is a value passed to the callback
[https://github.com/mafintosh/hyperdb#watcher--dbwatchfolderorkey-onchange](https://github.com/mafintosh/hyperdb#watcher--dbwatchfolderorkey-onchange)

File: test/masq.test.js (hunk 3 above)
Comment (sujeebant) [reply test/masq.test.js]: Yes we could remove the promise on mocked masq app

File: test/masq.test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (G-Ray) [reply test/masq.test.js]: Ah yeah, maybe we should edit the watch function later to get the keys which has changed, will see later, I created an issue in balcklog

File: test/masq.test.js (hunk 3 above)
Comment (sujeebant) [reply test/masq.test.js]: done !

File: test/masq.test.js (hunk 4 above)
Comment (sujeebant) [reply test/masq.test.js]: Promise.resolve() returns a Promise, I don't think we can set a watcher and then await the promise ?

File: test/masq.test.js (hunk 5 above)
Comment (sujeebant) [reply test/masq.test.js]: done

**Pull request merged**